src/ModalWithFooter.js

```diff
@@ -0,0 +1,59 @@
+// @flow
+
+import React from 'react';
+import Modal from './Modal';
+import classNames from 'classnames';
+import { Button } from 'react-bootstrap';
+import './ModalWithFooter.scss';
+
+type ModalWithFooterProps = {
+    show: boolean,
+    focusElementSelector: string,
+    focusOnCloseSelector: string,
+    ariaLabel?: string,
+    ariaLabelledById?: string,
+    ariaDescribedById?: string,
+    children?: any,
+    onClose: () => void,
+    buttonProperties: Array<{label: string, onClick: () => void, isPrimary: boolean}>
```

Comment:
Seems like we'd have to write less boilerplate if `isPrimary` could be optional, so that we'd only have to specify it for the primary button.

I do kind of wonder about this strategy WRT three button layouts, but as we only have 1 and 2 button modals at the moment, it's fine to have `isPrimary` for the primary and know that a secondary button omits that or has it set to false.